tests/Listener/MetaPalettesBuilderTest.php

```diff
@@ -39,32 +39,16 @@ class MetaPalettesBuilderTest extends TestCase
      */
     public function mockBuilder($dca)
     {
-        $builder = $this->getMockBuilder(MetaPalettesBuilder::class)
+        $framework = $this->getMockBuilder(ContaoFrameworkInterface::class)->getMock();
+        $builder   = $this->getMockBuilder(MetaPalettesBuilder::class)
             ->setMethods(array('loadDca'))
-            ->setConstructorArgs([new MetaPaletteParser()])
+            ->setConstructorArgs([$framework, new MetaPaletteParser()])
             ->getMock();
 
-        $reflection = new \ReflectionProperty(
-            'ContaoCommunityAlliance\DcGeneral\Contao\Dca\Builder\Legacy\DcaReadingDataDefinitionBuilder',
-            'dca'
-        );
+        $reflection = new \ReflectionProperty(MetaPalettesBuilder::class, 'dca');
         $reflection->setAccessible(true);
         $reflection->setValue($builder, $dca);
```

Comment:
We could now even skip this reflection setting and properly use the mocked adapter since that is what they were created for.
Pseudo code:
```php
$controller = $this->getMockBuilder(Controller::class)->getMock();
$controller->expects($this->once())->method('loadDataContainer')->with('tl_whatever')->willReturnCallback(function () use ($dca) {
    $GLOBALS['TL_DCA']['tl_whatever'] = $dca;
    return true;
})
$framework->expects($this->once())->method('getAdapter')->with(Controller::class)->willReturn($controller)
`` 

What do you think?

Reply:
Fully agree. That's what I actually wanted to do. Honestly I didn't know phpunits mockbuilder very well, that's why I kept the reflection hack.

I also was thinking about dropping the inherit method from the `Interpreter` as there is no difference between the current implementations. Furthermore we could drop the internal marked parsePalette from the parser interface. 

Do you see any future use case why we should keep it?